jsons/serializers/default_union.py

```diff
@@ -0,0 +1,30 @@
+from typing import Union
+from jsons._common_impl import get_class_name
+from jsons._compatibility_impl import get_union_params
+from jsons._dump_impl import dump
+from jsons.exceptions import JsonsError, SerializationError
+
+
+def default_union_serializer(obj: object, cls: Union, **kwargs) -> object:
+    """
+    Serialize an object to any matching type of the given union. The first
+    successful serialization is returned.
```

Comment:
That means the order you specified can change serialization value. For example:

```
@dataclass()
class Example:
     value1: Union[str, int]
     value2: Union[int, str]

jsons.dump(Example(10, 10)) -> {"value1": "10", "value2": 10}
```
Is that correct? If so, maybe we should iterate over each types in union and do `isinstance` to figure out?

Reply:
That is correct. The same goes for ``jsons.load`` with a ``Union``.

I'd rather not limit this with ``isinstance``, because there are cases in which you want to dump some instance with an "unrelated" type (e.g. ``jsons.dump('42', cls=int)``).

I really appreciatie your reviews by the way! 😃

Reply:
I see. You may consider documenting it then :)